opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java

```diff
@@ -0,0 +1,404 @@
+                        new EqArg(Instances.INSTANCE_DUE, start.addDuration(duration).getTimestamp()),
+                        new EqArg(Instances.INSTANCE_DURATION, durationMillis),
+                        new EqArg(Tasks.TZ, "UTC")
+                ))
+        ));
+    }
+
+
+    /**
+     * Having a table with a single task. Update task and check if instance updates accordingly.
+     */
+    @Test
+    public void testUpdateDue()
+    {
+        RowSnapshot<TaskLists> taskList = new VirtualRowSnapshot<>(new LocalTaskListsTable(mAuthority));
+        RowSnapshot<Tasks> task = new VirtualRowSnapshot<>(new TaskListScoped(taskList, new TasksTable(mAuthority)));
+        OperationsQueue queue = new BasicOperationsQueue(mClient);
+
+        DateTime start = DateTime.now();
+        DateTime due = start.addDuration(new Duration(1, 0, 1));
+        TimeZone timeZone = TimeZone.getDefault();
+
+        assertThat(new MultiBatch(
+                new Put<>(taskList, new NameData("list1")),
+                new Put<>(task, new TimeData(start, due))
+
+        ), resultsIn(queue,
+                new Assert<>(task, new TimeData(start, due)),
+                new PostOperationRelatedAssert<>(new InstanceTable(mAuthority), task, Instances.TASK_ID, new AllOf(
+                        new EqArg(Instances.INSTANCE_START, start.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_DUE, due.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_DURATION, due.getTimestamp() - start.getTimestamp()),
+                        new EqArg(Tasks.TZ, "UTC")
+                ))
+        ));
+
+        DateTime due2 = due.addDuration(new Duration(1, 0, 2));
+
+        assertThat(new SingletonBatch(
+                new Put<>(task, new TimeData(start, due2))
+
+        ), resultsIn(queue,
+                new Assert<>(task, new TimeData(start, due2)),
+                new PostOperationRelatedAssert<>(new InstanceTable(mAuthority), task, Instances.TASK_ID, new AllOf(
+                        new EqArg(Instances.INSTANCE_START, start.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_DUE, due2.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_DURATION, due2.getTimestamp() - start.getTimestamp()),
+                        new EqArg(Tasks.TZ, "UTC")
+                ))
+        ));
+    }
+
+
+    /**
+     * Having a table with a single task. Delete task and check if instance is deleted accordingly.
+     */
+    @Test
+    public void testInstanceDelete()
+    {
+        RowSnapshot<TaskLists> taskList = new VirtualRowSnapshot<>(new LocalTaskListsTable(mAuthority));
+        Table<Tasks> taskTable = new TaskListScoped(taskList, new TasksTable(mAuthority));
+        RowSnapshot<Tasks> task = new VirtualRowSnapshot<>(taskTable);
+        OperationsQueue queue = new BasicOperationsQueue(mClient);
+
+        assertThat(new MultiBatch(
```

Comment:
There is no need to assert this again if it already has been verified by a previous test. Just insert the rows.

Reply:
Removed that unnecessary assert.